### ISSUE TITLE
lock down versioning

### ISSUE THREAD

**Issue by mathematicalmichael** [mathematicalmichael/thesis#12]
make sure correct branch of BET is pointed to right now, before you bother updating any scripts.

Lock in the image tag for fenics, because `current` is being used right now, not a good idea given the unstable state of the fenics/dolfin project.

**Comment by mathematicalmichael** [comment]
CI is set up and working, though python scripts are only tested locally. 

BET is properly cloned

Still need to lock-in a specific tag, though there's an argument that with CI + Scheduling, it's not a problem.

Get the docker job running weekly, perhaps set up a second job that does a `sed` replacement for `latest` to specify a stable tag. This way we test "locked in" version + "latest". CI tests are using the docker-hub versions, so in some sense as long as those aren't written over, we should be safe with the existing workflow (no need to separate out jobs). 

In summary:
- dockerfiles refer to `latest`
- CI pipeline pulls static images for testing. 

- A test that builds dockerfiles with latest and tries to also build examples is needed. Since, it may be one thing for a build to be successful for an environment, but the versions of libraries may break things. This job can run perhaps only on PRs into master, scheduled to run weekly as an end-to-end test.

**Comment by mathematicalmichael** [comment]
`mpilosov/BET/archive` is the branch to clone to recreate masters branch, include as part of CI pipeline (separate workflow for now, eventually figure out which images we can delete). may be smart to create a version of the `run.sh` file in `estimateDist` that runs only exactly what we need as far as figure-generation goes. I would say this is pretty low priority, since I know it can be done, it's just a matter of time.

**Comment by mathematicalmichael** [comment]
there are a bunch of notes about this in trello now
